File: clients/app/ios/PetSitter/Presentation/Auth/LoginScreen.swift
import SwiftUI

struct LoginScreen: View {
    @Environment(AuthStore.self) private var authStore
    @State private var viewModel = LoginViewModel()

    var body: some View {
        @Bindable var vm = viewModel

        ScrollView {
            VStack(alignment: .leading, spacing: 0) {
                Spacer().frame(height: 48)

                Text("로그인")
                    .font(AppTypography.t1)
                    .foregroundStyle(AppColors.textPrimary)

                Spacer().frame(height: 32)

                AppTextField(
                    label: "이메일",
                    placeholder: "email@example.com",
                    text: $vm.email,
                    keyboardType: .emailAddress,
                    errorMessage: viewModel.emailError
                )

                Spacer().frame(height: 16)

                AppTextField(
                    label: "비밀번호",
                    placeholder: "6자 이상",
                    text: $vm.password,
                    isSecure: true,
                    errorMessage: viewModel.passwordError
                )

                if let error = viewModel.serverError {
                    Spacer().frame(height: 8)
                    Text(error)
                        .font(AppTypography.caption)
                        .foregroundStyle(AppColors.danger)
                }

                Spacer().frame(height: 24)

                AppButton(title: "로그인", isLoading: viewModel.isLoading) {
                    viewModel.submitted = true
                    guard viewModel.isValid else { return }
                    Task {
                        viewModel.isLoading = true
                        viewModel.serverError = nil
                        do {
                            try await authStore.login(email: viewModel.email, password: viewModel.password)
                        } catch let error as NetworkError {
                            viewModel.serverError = error.errorDescription
                        } catch {
                            viewModel.serverError = "서버 오류가 발생했습니다"
                        }
                        viewModel.isLoading = false
                    }
                }

                Spacer().frame(height: 16)

                HStack {
                    Spacer()
                    Text("계정이 없으신가요?")
                        .font(AppTypography.b2)
                        .foregroundStyle(AppColors.textSecondary)
                    NavigationLink("회원가입") {
                        RegisterScreen()
                    }
                    .font(AppTypography.b2)
                    .foregroundStyle(AppColors.primary)
                    Spacer()
                }
            }
            .padding(.horizontal, 24)
            .padding(.bottom, 16)
        }
        .background(AppColors.white)
    }
}

File: clients/app/android/app/src/main/java/com/petsitter/presentation/auth/LoginScreen.kt
package com.petsitter.presentation.auth

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.compose.foundation.layout.WindowInsets
import androidx.compose.foundation.layout.systemBars
import androidx.compose.foundation.layout.windowInsetsPadding
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.petsitter.ui.components.AppButton
import com.petsitter.ui.components.AppTextField
import com.petsitter.ui.theme.AppColors
import com.petsitter.ui.theme.AppTypography

// LoginScreen: View 대응 — 폼 렌더링만, 로직은 LoginViewModel + AuthViewModel에서
@Composable
fun LoginScreen(
    onNavigateToRegister: () -> Unit,
    authViewModel: AuthViewModel,
    loginViewModel: LoginViewModel = hiltViewModel(),
) {
    val state by loginViewModel.state.collectAsStateWithLifecycle()

    Column(
        modifier = Modifier
            .fillMaxSize()
            // status bar + nav bar safe area 처리 (Scaffold 없는 화면에서 직접 명시)
            .windowInsetsPadding(WindowInsets.systemBars)
            .verticalScroll(rememberScrollState())
            .padding(horizontal = 24.dp, vertical = 32.dp),
    ) {
        Spacer(modifier = Modifier.height(48.dp))

        Text(
            text = "로그인",
            style = AppTypography.t1,
            color = AppColors.textPrimary,
        )

        Spacer(modifier = Modifier.height(32.dp))

        AppTextField(
            value = state.email,
            onValueChange = loginViewModel::onEmailChange,
            label = "이메일",
            placeholder = "email@example.com",
            errorMessage = state.emailError,
            keyboardType = androidx.compose.ui.text.input.KeyboardType.Email,
        )

        Spacer(modifier = Modifier.height(16.dp))

        AppTextField(
            value = state.password,
            onValueChange = loginViewModel::onPasswordChange,
            label = "비밀번호",
            placeholder = "6자 이상",
            errorMessage = state.passwordError,
            isPassword = true,
        )

        if (state.serverError != null) {
            Spacer(modifier = Modifier.height(8.dp))
            Text(
                text = state.serverError!!,
                style = AppTypography.caption,
                color = AppColors.danger,
            )
        }

        Spacer(modifier = Modifier.height(24.dp))

        AppButton(
            text = "로그인",
            isLoading = state.isLoading,
            onClick = {
                if (loginViewModel.validate()) {
                    loginViewModel.setLoading(true)
                    loginViewModel.setServerError(null)
                    authViewModel.login(
                        email = state.email,
                        password = state.password,
                        onError = { msg ->
                            loginViewModel.setLoading(false)
                            loginViewModel.setServerError(msg)
                        },
                    )
                }
            },
        )

        Spacer(modifier = Modifier.height(16.dp))

        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.Center,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Text(
                text = "계정이 없으신가요?",
                style = AppTypography.b2,
                color = AppColors.textSecondary,
            )
            TextButton(onClick = onNavigateToRegister) {
                Text(
                    text = "회원가입",
                    style = AppTypography.b2,
                    color = AppColors.primary,
                )
            }
        }
    }
}
